### PR TITLE
Support obsm embeddings in the Loader

### PR DESCRIPTION
**Pull request by moinfar**
## Summary

Adds support for loading dense `obsm` arrays (e.g. a scVI embedding `"X_emb"`) through the mini-batch `Loader`, yielded row-aligned with `X`.

- New `obsm_keys` argument on `Loader.add_adata` / `add_adatas` (and an `obsm` argument on `add_dataset` / `add_datasets`) selects which obsm keys to load — only the requested keys are read, keeping I/O minimal.
- Backward compatible: on-disk store format is untouched; existing calls are unchanged. The only observable difference is the extra `obsm=None` key in each yielded batch.
- Validation raises clear errors for: non-dense obsm, mismatched n_obs, and inconsistent keys / feature-shape / dtype across datasets. obsm dtype/shape is required to be homogeneous across datasets (kept simple; can be relaxed later).

## Use of AI agents
This change was developed with AI assistance (Claude Code). I briefly reviewed all changes (the implementation and tests were authored via the tool and reviewed by me). The correctness of row alignment, concurrency, and edge-case handling was additionally checked by a second run of an AI agent.

### CODE REVIEW

**Comment by moinfar** [comment]
Is related to #91 and #102

**Comment by selmanozleyen** [comment]
Thanks! this will be done in the following PR's after https://github.com/scverse/annbatch/pull/251, because we need to warn the users about the changing default behaviour first.

**Comment by moinfar** [comment]
Thanks a lot.

File: src/annbatch/loader.py [hunk 1]
```diff
@@ -384,52 +385,67 @@ def use_collection(
     def add_adatas(
         self,
         adatas: list[ad.AnnData],
+        *,
+        obsm_keys: list[str] | None = None,
```
Comment (ilan-gold) [inline src/annbatch/loader.py:389]: Yeah, the goal is not to have to special-case this but to use https://anndata.scverse.org/en/stable/accessors.html

i.e.,
```python
Loader(...).add_adatas(adatas, acc={A.obsm["pca"], A.layers["counts"]})

Comment (moinfar) [reply src/annbatch/loader.py:389]: This accessor concept is really interesting.
Looking forward to seeing it in the annbatch interface.
